cmd/kubelet/app/server.go

```diff
@@ -722,7 +723,13 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.Dependencies) (err error) {
 	// If systemd is used, notify it that we have started
 	go daemon.SdNotify(false, "READY=1")
 
-	<-done
+	select {
```

Comment:
so `done`  is a lockfile-based method for telling the kublet to shut down, and stopCh is a signal-based mechanism? I'm surprised we weren't already passing `done` down into RunKubelet to get graceful shutdowns

Reply:
should we be calling `SdNotify(false, "STOPPING=1")` as part of shutdown?

do we have to wait until RunKubelet has finished responding to the closed stopCh before we return?

Reply:
> should we be calling SdNotify(false, "STOPPING=1") as part of shutdown?

That I don't know; I'm speculating that we don't see this in the tree at all because we didn't really gracefully shutdown before.

Reply:
You don't have to call the shutdown signal.